components/python/virtualenv/Makefile

```diff
@@ -0,0 +1,61 @@
+#
+# This file and its contents are supplied under the terms of the
+# Common Development and Distribution License ("CDDL)". You may
+# only use this file in accordance with the terms of the CDDL.
+#
+# A full copy of the text of the CDDL should have accompanied this
+# source. A copy of the CDDL is also available via the Internet at
+# http://www.illumos.org/license/CDDL.
+#
+
+#
+# Copyright (c) 2019 Tim Mooney
+#
+
+include ../../../make-rules/shared-macros.mk
+
+COMPONENT_NAME=		virtualenv
+COMPONENT_VERSION=	16.2.0
+COMPONENT_FMRI=		library/python/virtualenv
+COMPONENT_SUMMARY=	'A tool to create isolated Python environments'
+COMPONENT_PROJECT_URL=	https://virtualenv.pypa.io/en/latest/
+COMPONENT_CLASSIFICATION=Development/Python
+COMPONENT_SRC=		$(COMPONENT_SRC_NAME)-$(COMPONENT_VERSION)
+COMPONENT_ARCHIVE=	$(COMPONENT_SRC).tar.gz
+COMPONENT_ARCHIVE_HASH=	\
+    sha256:448def1220df9960e6d18fb5424107ffb1249eb566a5a311257860ab6b52b3fd
+COMPONENT_ARCHIVE_URL=	https://github.com/pypa/virtualenv/archive/$(COMPONENT_VERSION).tar.gz
+COMPONENT_LICENSE=	MIT
+COMPONENT_LICENSE_FILE=	virtualenv.license
+
+PYTHON_VERSIONS=        2.7 3.4 3.5
+
+include $(WS_TOP)/make-rules/prep.mk
+include $(WS_TOP)/make-rules/setup.py.mk
+include $(WS_TOP)/make-rules/ips.mk
+
+COMPONENT_TEST_DIR =	$(SOURCE_DIR)
+COMPONENT_TEST_ARGS =	setup.py test
+
+# Replace "#!/usr/bin/env ..." shebang lines with properly versioned ones.
+COMPONENT_POST_INSTALL_ACTION += \
+    $(FIND) $(PROTOUSRLIBDIR)/python$(PYTHON_VERSION) -name *.py \
+        -exec $(GSED) -i -e 's|env python|python$(PYTHON_VERSION)|' "{}" \; ;
+
+COMPONENT_POST_INSTALL_ACTION += \
+        (cd $(PROTOUSRBINDIR); $(MV) virtualenv virtualenv-$(PYTHON_VERSION)) ;
+
+# common targets
+build:		$(BUILD_32_and_64)
```

Comment:
Can you use BUILD_NO_ARCH? This seems to be correct as component doesn't produce binary files